src/Helpers.hpp

```diff
@@ -0,0 +1,75 @@
+#ifndef __RADAR_BASE_HELPERS_HPP__
+#define __RADAR_BASE_HELPERS_HPP__
+
+#include "EchoToImageLUT.hpp"
+#include "Radar.hpp"
+#include <base/Angle.hpp>
+#include <base/Float.hpp>
+#include <base/Time.hpp>
+
+#include <vector>
+namespace radar_base {
+    namespace helpers {
+
+        /**
+         * Function to be iterated inside each sweep, its inputs are the radar echo
+         * itself, the current local echo sweep index as well as the global sweep angular
+         * position.
+         */
+        using sweepFunction = std::function<void(Radar const&, int, int)>;
+
+        /**
+         * Iterates over the sweeps of a radar echo and applies a
+         * user-defined function to each sweep. This function should take the radar echo,
+         * the sweep local index and the sweep global angular position as input.
+         *
+         * @param radar_echo
+         * @param yaw_correction
+         * @param lambda_function
+         */
+        void iterateOverSweepsAndApplyFunction(Radar const& radar_echo,
+            base::Angle const& yaw_correction,
+            sweepFunction const& lambda_function);
+
+        /**
+         * Resets the lookup table given new parameters.
+         *
+         * @param num_angles
+         * @param sweep_size
+         * @param beam_width
+         * @param output_image_size
+         * @param m_lut
+         */
+        void updateLookUpTable(int num_angles,
+            int sweep_size,
+            float beam_width,
+            int output_image_size,
+            std::unique_ptr<radar_base::EchoToImageLUT>& m_lut);
+
+        /**
+         * Resets the lookup table if the radar echo configuration is different from
+         * what is expected, i.e if differing range, sweep size or the number of sweeps
+         * per revolution.
+         *
+         * @param num_angles
+         * @param sweep_size
+         * @param radar_range
+         * @param radar_echo
+         * @param beam_width
+         * @param output_image_size
+         * @param m_lut
+         *
+         * @return true if it reset the lookup table
+         * @return false if it kept the same lookup table
+         */
+        bool updateLookUpTableIfNeeded(int num_angles,
```

Comment:
I think that when LUT parameters correspondence is made by the LUT class, this method will become unnecessary